pkg/synchronization/controller.go

```diff
@@ -665,6 +705,40 @@ func pagerDutyForEnvType(pagerduty *creator_v1.PagerDutyMetadata, envType voyage
 	}
 }
 
+// filterOpsgenieIntegrationsByEnv filters a given list of integrations by the EnvType
+func filterOpsgenieIntegrationsByEnv(integrations []opsgenie.Integration, envType voyager.EnvType) ([]opsgenie.Integration, error) {
+	if len(integrations) == 0 {
+		return nil, nil // Not an error as Opsgenie is optional
+	}
+
+	filtered := make([]opsgenie.Integration, 0, 4)
+
+	for _, integration := range integrations {
+
+		if integration.EnvType == opsgenie.EnvTypeGlobal {
+			filtered = append(filtered, integration)
+			continue
+		}
+
+		switch envType {
+		case voyager.EnvTypeStaging:
+			if integration.EnvType == opsgenie.EnvTypeStaging {
+				filtered = append(filtered, integration)
+			}
+		case voyager.EnvTypeProduction:
+			if integration.EnvType == opsgenie.EnvTypeProd {
+				filtered = append(filtered, integration)
+			}
+		default:
```

Comment:
I would prefer to have `case voyager.EnvTypeDev` here and throw an error in the `default:` section flagging an unexpected/unknown envType.